### DISCUSSION TITLE
Ship Systems - Life Support

### DISCUSSION THREAD

**Discussion by Froginator45** [Thorium-Sim/thorium-nova#231] [Feature Proposals]
_For more information about the ship systems discussions, see https://github.com/Thorium-Sim/thorium-nova/discussions/224_

# Life Support
In space travel and exploration, life support is the primary component of the space vessel. It consists of various systems with their redundancies that work together to provide the basics of the crew’s survival. A life support system addresses basic survival needs including food, water, and air.

### ISS
The typical astronaut crewmember aboard the ISS (of average body size) requires a combined 5 kg (11 lbs) of food, water, and air per day; an almost identical weight is expelled from the body in the form of carbon dioxide, and liquid and solid waste. The mass breakdown of these metabolic parameters is as follows: 0.84 kg (1.9 lb) of oxygen, 0.62 kg (1.4 lb) of food, and 3.54 kg (7.8 lb) of water consumed, converted through the body's physiological processes to 0.11 kg (3.9 oz) of solid wastes, 3.89 kg (8.6 lb) of liquid wastes, and 1.00 kg (2.20 lb) of carbon dioxide produced. 

The life support systems must address these needs, supplying the necessary intake and disposing (or recycling) of the consequential outputs.
### Thorium
Generally, life support effectiveness would affect crew moral (effectiveness). It would only be after a certain amount of time that you would start to lose crew members due to a lack of one of the life support systems. Most, if not all, of these systems would be automated and only bring awareness to the flight director if conditions reached a critical point (i.e. water reserves have been depleted for some time, radiation levels are past a certain point, fires have not been handled quickly, etc.). 

An overall crew moral screen might be useful for the captain to determine what could be done to help increase crew moral and see how effective the crew currently is. Exceptionally low moral might result in strikes.
## 7 Basic Components of a Life Support System
At its most basic level, life support systems must address seven core human needs. The necessary components of a life support system address:
1.	Potable water supply trough water reclamation
2.	Food supply
3.	Breathable cabin air supply (oxygen generation system)
4.	Air pressure control (maintained at 101.kPa)
5.	Air temperature regulation through a heat exchanger
6.	Human waste management and waste disposal
7.	Fire detection and suppression

To address the above needs, the current NASA ECLSS system standards include the following components of a life support system:

- Crew compartment cabin pressurization
- Cabin air revitalization
- Water coolant loop system
- Active thermal control system
- Supply and wastewater
- Water collection system
- Wastewater tank
- Airlock support
- Extravehicular mobility units
- Crew altitude protection system
- Radioisotope thermoelectric generator cooling and gaseous nitrogen purge for payloads

## Oxygen/Air Pressure
Air pressure can also have a detrimental effect to crew. Humans can’t survive in pressures above 2.5 bar due to the partial pressure of oxygen. At pressures below 121 millibars, hypoxia begins to set in. and below 61.8 millibars, blood stops to flow. 1.016 bars is the optimal pressure for humans.
### ISS

- Crew compartment cabin pressurization
- Cabin air revitalization
- Airlock support

Aboard the ISS, the Oxygen Generation System produces oxygen for breathing air, as well as replaces oxygen lost as a result of experiment use, airlock depressurization, module leakage, and carbon dioxide venting. The OGS primarily consists of the Oxygen Generation Assembly (OGA) and a Power Supply Module. Oxygen is generated at a selectable rate and is capable of operating continuously and cyclically.

Part of the Oxygen Generation System involves removing trace contaminants produced by electronics, plastics, and human off-gassing, including carbon dioxide exhaled by the crew during normal respiration. On the ISS, trace contaminants are removed by flowing cabin air through three separate units including an activated charcoal bed, a catalytic oxidizer and a lithium hydroxide bed. Carbon dioxide is removed using molecular sieves. Many of these systems will require “filters” to be replaced as part of normal maintenance. 

The Oxygen Generation System produces oxygen for the crew to breathe. The system consists of the oxygen generation assembly and the carbon dioxide reduction assembly. The oxygen generation assembly is composed of the cell stack, which electrolyzes, or breaks apart, water provided by the Water Recovery System, yielding oxygen and hydrogen as byproducts. The oxygen is delivered to the cabin atmosphere while the hydrogen is either vented into space or fed to the carbon dioxide reduction assembly. The assembly uses that hydrogen along with carbon dioxide exhaled by the crew in a Sabatier reactor. The byproducts of that process are methane (which is released into space) and water for use by the crew.
### Thorium
The crew could monitor the oxygen tank pressures and current “filter useable-life” of the various Oxygen Generation systems and trace contaminant filters. Oxygen would be used at a specific rate due to the # of crew aboard. It would also be used for any experiments that the crew may conduct. Certain probes might need a specified amount of oxygen to complete their functions and therefore will empty the O2 tanks slightly upon creation. A filter screen might also show current filter efficiencies and need to be replaced periodically. This would affect the number of contaminants in the cabin air (and eventually the crew’s effectiveness). There could also be a screen showing air purity and/or contaminant concentration. 

As a topic related to oxygen generation; the hydrogen produced from the water electrolization could also be used to power hydrogen engines. It is possible that many probes and auxiliary systems use hydrogen as their fuel. If this were the case, then a hydrogen tank pressure screen would also be useful for the crew. 

Oxygen reserves could be replenished while docked at a station. Additional emergency supply tanks, filters, and pressure regulators could also be purchased while docked. Filters might be upgradeable and result in requiring less frequent maintenance. 

Minigame possibility: basic pipe game to route air to a new section of the ship. Could be presented as a new screen every-time they needed to reroute or as a static screen that normally allows air to be provided everywhere but with damage, certain pipe sections could be rendered unusable, and therefore need to be bypassed. And, over time, repairs to the system might need to take place to fix certain sections of pipe in the minigame and give the crew more possibilities of air supply. Upgrades might also be convenient by adding additional useful pipe sections or additional supply locations.
### ISS

- Supply and wastewater
- Water collection system
- Wastewater tank

The Water Recovery System aboard the ISS, provides clean water by recycling crewmember urine, cabin humidity condensate, and Extra Vehicular Activity (EVA) wastes. The reclaimed water must meet stringent purity standards before it can be utilized to support the crew, laboratory animals, EVA, and payload activities. The WRS consists of a Urine Processor Assembly (UPA) and a Water Processor Assembly (WPA).

Water is recycled from urine and dehumidifiers, typically with about 90% efficiency. That’s better than ever, but we need to get to virtually 100% recycling before we can venture long distance.

Each astronaut needs about a gallon of water per day for consumption, food preparations and hygiene—brushing teeth and shaving.
### Thorium
The crew could monitor the water tank levels and current “filter useable-life” of the various Water Generation systems. Water would be used at a specific rate due to the # of crew aboard. It would also be used for any experiments that the crew may conduct. Certain probes might need a specified amount of water to complete their functions and therefore will empty the H2O tanks slightly upon creation. As water reserves lower, the crew may decide to implement rations. This could affect crew moral (effectiveness). A filter screen might also show current filter efficiencies and need to be replaced periodically. 

Water reserves could be replenished while docked at a station. Additional emergency supply tanks and filters could also be purchased while docked. Filters might be upgradeable and result in requiring less frequent maintenance. 

Minigame possibility: basic pipe game to route water to a new section of the ship (like the minigame described in the oxygen section above).
### ISS
Food is even more difficult to recycle, as farming is a multi-stage process, growing seasons take time, and a balanced diet is essential. For simplicity and reliability, the ISS receives virtually all of the astronauts’ food via regular deliveries from Earth. To ensure long shelf life and to minimize the chance of food poisoning, meals are dehydrated, irradiated, thermo-stabilized and/or canned. Preparation is kept simple, with a water dispenser and warming ovens. Only rarely, as a kindness from the support team, can a few items of fresh fruit or vegetables be sent to the crew, added as late stowage on a resupply ship. 
### Thorium
I generally see food being supplied by food replicators. A lack of food would also affect the crew moral (effectiveness). On-board hydroponics might also be present as the source of food that could be affected by the water reserves. Hydroponic systems can quickly deteriorate with a lack of water and lose an entire crop of food. Fresh food supply runs from local suppliers could be part of various missions with fresh food giving a moral boost to crew.
## Temperature

- Water coolant loop system
- Active thermal control system

Air temperature and water temperature are both important to a crew’s life support. In the cold, crew can become lethargic and spending more time trying to stay warm. In the heat, crew will feel exhausted and not want to work. Water that is too hot would be unusable by various needs of the crew and ship functions. Water that gets too cold could freeze and damage various systems by bursting pipes, etc.  Air temperature could be considered room-by-room with temperature gradients between the rooms being realistic based on intermediate bulkhead door statuses. 
## Safety Systems

- Radiation
- Fire Suppression

Various safety systems, while not being directly responsible for crew life support, are essential in the case of an emergency or during long missions (radiation poisoning). Basic ship structure (hull integrity) should generally take care of basic environmental radiation. During, increased radiation situations, the shields would be required. Fire suppression systems would generally be automated but could cause cascading failures if they don’t respond quickly and with enough volume. 
Other internal safety systems might include:

- Bio-contaminant detection (disease control)
- Safety bulkheads 
  - Potential for power to be cut to each bulkhead (for emergency opening or to slow down intruders)
- Power grounding pathways

## Conveniences

- Artificial Gravity
- Holodeck
- Turbolifts
- Brig Force fields

Artificial gravity (since it can’t affect the real crew) would mostly affect the AI crew’s effectiveness. It could also possibly cause damage in storage areas and result in loss of useable consumables (replacement parts, building materials, food, water, etc.). An interesting aspect of artificial gravity is if the gravity emitters are only strong enough to cover sections of the ship. This could then cause more localized effects to the crew and storage. And the artificial gravity in the bridge could always seem to never be affected by the game’s “RNG”.

Holodeck would mostly be used as a crew moral boost. The potential power consumption of the holodecks would reflect the # of holodecks on-board.

Turbolifts would increase crew travel time (and therefore, their overall effectiveness).

 ---
Am I missing anything? Comments, suggestions, feedback, or alternate proposals welcome!

**Comment by alexanderson1993** [comment]
Excellent proposal. I love a lot of these ideas. 

> Generally, life support effectiveness would affect crew moral 

Sometime soon, I'll start a discussion about all of the different properties of crew members that affect their behavior. This will work really nicely with that.

> I generally see food being supplied by food replicators.

I don't disagree, but the existence of food replicators implies a lot. The assumption is that this device can turn raw energy into organic material that is safe for human consumption. If that's the case, can it replicate water, or air, or parts, or anything else?

Replicators as a concept kind of magic away some of the difficult parts of life support, like managing water levels or even waste (since you likely can magically turn waste back into energy, or at very least disintegrate it).  At that point, you basically just need fuel, reactors to generate energy, and working replicators to generate whatever you need. 

And, if we want to have transporters of some kind, that also implies a matter-energy transfer, so the crew does have the technology already.

All that said, I think we should keep the proposal as-is and if necessary explain that there are limitations to the replicator technology that make it not work for water or air. Or that it's much more energy efficient to electrolyze water to produce oxygen, or to combust hydrogen to produce water. Maybe lunchtime is an especially taxing time for the reactors, since so many people are replicating food all at once - that would be a fun little irony 😆

> Air temperature could be considered room-by-room with temperature gradients between the rooms being realistic based on intermediate bulkhead door statuses.

This implies an HVAC duct utility layer, which I'm all in favor of

---

One thought that I want to address is how this fits into the broader set of controls. In Thorium Classic, I never implemented an Environment or Life Support card, because it rarely has a direct connection to the story. Life support, while vital to the functioning of the ship, doesn't directly contribute to the story the same way that engines or weapons do. I suppose, like everything else, life support can be written into the story in a meaningful way, but it will be important to avoid unnecessary busywork.

So there's a balance between making realistic systems and fun gameplay, and we need to identify the intersection. There are a few ways I see that with this proposal:

- Keeping the NPC crew happy and engaged so they perform their tasks well and the ship keeps functioning.
- Using life support strategically, like pumping nitrogen into a room with intruders so they pass out.
- Putting the crew in crisis mode when certain life support systems are failing, giving the crew a chance to work together to make them work again.
- Specific story tasks, like making specific environment adjustments to a room so alien diplomats will be comfortable there.

If anyone has other ideas about how best to avoid busywork and integrating life support with the story, feel free to share them.

**Comment by Ryan-Anderson-PhD** [comment]
So... here's a thought.

If life support is severely damaged, what if we had the option of placing some crew members in cryogenic pods to reduce oxygen consumption, etc. until the function is restored. It could save lives, but would reduce the effectiveness and efficiency of the crew. 

You could also adjust artificial gravity in the docking bays/shuttle bays, turning it on for maintenance and preparation and off for launches. 

You could have water filtration and reclamation units, air filters, etc. Each one of these would need maintenance and inspection. It would also give something for security to do: creating patrols that check on vital life support systems (and other ship systems) and even placing guards at very crucial points.
